lib/actors/registry/actor_registry.ex

```diff
@@ -81,11 +82,47 @@ defmodule Actors.Registry.ActorRegistry do
     {:reply, :ok, state}
   end
 
+  @impl true
+  def handle_cast({:register_invocation_request, actor, request}, state) do
+    updated_hosts =
+      actor
+      |> StateHandoff.get()
+      |> Kernel.||([])
+      |> Enum.map(fn host ->
+        invocations = (host.opts[:invocations] || []) ++ [request]
+
+        opts = Keyword.put(host.opts, :invocations, invocations)
+        %{host | opts: opts}
+      end)
+
+    StateHandoff.set(actor, updated_hosts)
+
+    {:noreply, state}
+  end
+
+  def handle_cast({:remove_invocation_request, actor, request}, state) do
+    updated_hosts =
+      actor
+      |> StateHandoff.get()
+      |> Kernel.||([])
+      |> Enum.map(fn host ->
+        invocations = host.opts[:invocations] || []
+        invocation = Enum.find(invocations, &(&1 == request))
+        invocations = invocations -- [invocation]
+
+        opts = Keyword.put(host.opts, :invocations, invocations)
+        %{host | opts: opts}
+      end)
+
+    StateHandoff.set(actor, updated_hosts)
+
+    {:noreply, state}
+  end
+
   @impl true
   def terminate(_reason, _state) do
     Logger.debug("Stopping ActorRegistry...")
-    _node = Node.self()
-    # StateHandoff.clean(node)
+    StateHandoff.clean(Node.self())
```

Comment:
should we just call clean here?

Reply:
I don't think so, as statehandoff is an external module, I think it's interesting to pass Node. By the way we have a serious bug in this routine because it cleans all the actors of a type that was registered by the node and not just the actors of that node that is being cleaned.

Reply:
But we can handle the bug in another commit or PR